### PR TITLE
dbcp general formular; duplicate twin

### PR DESCRIPTION
**Pull request by Tschuck**
- add general dbcp edit formular
- add correct image urls for dbcp edit
- add twin duplicate
- [CORE-599]

[CORE-599]: https://evannetwork.atlassian.net/browse/CORE-599

### CODE REVIEW

File: core/digital-twin-lib/src/i18n/de.json [hunk 1]
```diff
@@ -7,11 +7,13 @@
       },
       "description": "Beschreibung speichern...",
       "twin": {
+        "create": "Digitaler Zwilling wird erstellt...",
```
Comment (karladler) [inline core/digital-twin-lib/src/i18n/de.json]:
```suggestion
        "create": "Digitaler Zwilling wird erstellt ...",
```

File: core/digital-twin-lib/src/i18n/en.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
       },
       "description": "Saving description...",
       "twin": {
+        "create": "Creating Digital Twin...",
```
Comment (karladler) [inline core/digital-twin-lib/src/i18n/en.json]:
```suggestion
        "create": "Creating Digital Twin ...",
```

File: dapps/digital-twin-detail/src/components/DigitalTwinDetail.ts [hunk 3]
```diff
@@ -21,18 +21,22 @@ import Component, { mixins } from 'vue-class-component';
...
    */
-  loading = true;
+  duplicating = false;
   exporting = false;
   favoriteLoading = false;
+  loading = true;
 
+  /**
+   * Is the current twin a favorite?
+   */
```
Comment (karladler) [inline dapps/digital-twin-detail/src/components/DigitalTwinDetail.ts]: we once should talk about the sense of redundant comment blocks ... it's really obvious actually what this var represents.

File: dapps/digital-twin-detail/src/components/DigitalTwinDetail.ts [hunk 4]
```diff
@@ -83,10 +102,56 @@ export default class DigitalTwinDetailComponent extends mixins(EvanComponent) {
...
+  async createTwinDuplicate(): Promise<void> {
+    this.duplicating = true;
+    
+    const { description, image }: any = this.dbcpForm.getDescription();
+    // start twin duplicate dispatcher
+    await dispatchers.twinCreateDispatcher
+      .start(this.getRuntime(), {
+        description: { description },
+        twinImage: image,
+        twinTemplate: this.$store.state.twin.contractAddress,
```
Comment (karladler) [inline dapps/digital-twin-detail/src/components/DigitalTwinDetail.ts]: shouldn't this be an object?

File: dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue [hunk 5]
```diff
@@ -58,40 +58,107 @@
...
+                    class="light"
+                    :title="'_twin-detail.data.context-menu.export-template' | translate"
+                    showBackdrop="true"
+                    type="default">
+                    <div class="d-flex h-100 align-items-center justify-content-center">
+                      <div v-if="exporting">
+                        <evan-loading />
+                        <div class="mt-3 text-center">
+                          <h4>
+                            {{ `_twin-detail.data.context-menu.exporting-twin` | translate }}
```
Comment (karladler) [inline dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue]:
```suggestion
                            {{ '_twin-detail.data.context-menu.exporting-twin' | translate }}
```

File: dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue [hunk 6]
```diff
@@ -58,40 +58,107 @@
...
+                        </div>
+                      </div>
                       <evan-success v-else />
+                    </div>
+
+                    <template slot="footer">
+                      <div class="d-flex">
+                        <evan-button
+                          type="secondary"
+                          :label="$t('_evan.cancel')"
```
Comment (karladler) [inline dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue]:
```suggestion
                          :label="'_evan.cancel' | translate"
```

File: dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue [hunk 7]
```diff
@@ -58,40 +58,107 @@
...
+                          @click="exportModal.hide()"
+                          class="mr-3"
+                        />
+                        <span class="mx-auto" />
+                        <evan-button
+                          type="primary"
+                          :disabled="exporting"
+                          :label="$t('_twin-detail.data.context-menu.download-template')"
```
Comment (karladler) [inline dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue]: translation function or filter? We should use only one style in .vue template

File: dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue [hunk 8]
```diff
@@ -58,40 +58,107 @@
...
-                  <evan-modal ref="exportModal">
-                    <template v-slot:header>
-                      <h5 class="modal-title">
-                        {{ `_twin-detail.data.context-menu.export-template` | translate }}
-                      </h5>
-                    </template>
-
-                    <template v-slot:body>
-                      <evan-loading v-if="exporting" />
+                  <evan-swipe-panel
```
Comment (karladler) [inline dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue]: shall we outsource the swipe-panels to own components?

File: dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue [hunk 9]
```diff
@@ -58,40 +58,107 @@
...
                     </template>
-                  </evan-modal>
+                  </evan-swipe-panel>
                 </div>
 
                 <evan-profile-picture
                   class="twin-avatar"
                   type="device"
-                  :src="'https://via.placeholder.com/96'"
+                  :src="$store.state.twin.description.imgSquare"
```
Comment (karladler) [inline dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue]: is it sure that all keys always exists? Maybe you can wrap it into a getImageSrc function which handles missing fields smoothly in TS.

File: dapps/digital-twin-detail/src/components/DigitalTwinDetail.ts (hunk 3 above)
Comment (Tschuck) [reply dapps/digital-twin-detail/src/components/DigitalTwinDetail.ts]: True story for this parameter. Often i miss some small comments for anything.

File: dapps/digital-twin-detail/src/i18n/de.json [hunk 10]
```diff
@@ -2,18 +2,20 @@
   "_twin-detail": {
     "data": {
       "context-menu": {
-        "export-template": "Als Template exportieren",
-        "download-template": "Herunterladen"
+        "download-template": "Herunterladen",
+        "duplicate-twin": "Duplizieren",
+        "duplicating-twin": "Duplikat wird erstellt...",
```
Comment (karladler) [inline dapps/digital-twin-detail/src/i18n/de.json]:
```suggestion
        "duplicating-twin": "Duplikat wird erstellt ...",
```

File: dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue (hunk 9 above)
Comment (Tschuck) [reply dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue]: It's really sure, that this path exists. It is loaded previously.

File: dapps/digital-twin-detail/src/i18n/de.json [hunk 11]
```diff
@@ -2,18 +2,20 @@
...
+        "export-template": "Als Vorlage exportieren",
+        "exporting-twin": "Vorlage wird erstellt..."
```
Comment (karladler) [inline dapps/digital-twin-detail/src/i18n/de.json]:
```suggestion
        "exporting-twin": "Vorlage wird erstellt ..."
```

File: dapps/digital-twin-detail/src/i18n/en.json [hunk 12]
```diff
@@ -2,8 +2,11 @@
   "_twin-detail": {
     "data": {
       "context-menu": {
+        "download-template": "Download",
+        "duplicate-twin": "Duplicate",
+        "duplicating-twin": "Creating duplicate...",
```
Comment (karladler) [inline dapps/digital-twin-detail/src/i18n/en.json]:
```suggestion
        "duplicating-twin": "Creating duplicate ...",
```

File: dapps/digital-twin-detail/src/i18n/en.json [hunk 13]
```diff
@@ -2,8 +2,11 @@
...
         "export-template": "Export as template",
-        "download-template": "Download"
+        "exporting-twin": "Creating template..."
```
Comment (karladler) [inline dapps/digital-twin-detail/src/i18n/en.json]:
```suggestion
        "exporting-twin": "Creating template ..."
```

File: dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue (hunk 8 above)
Comment (Tschuck) [reply dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue]: Moved the swipe panels to a seperated `TwinInteractions` component.

File: dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue (hunk 7 above)
Comment (Tschuck) [reply dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue]: I would prefer to use translate pipe when it's possible.

**Pull request merged**